src/fmpz/rand.c

```diff
@@ -64,26 +67,27 @@ fmpz_randm(fmpz_t f, flint_rand_t state, const fmpz_t m)
 
 void fmpz_randprime(fmpz_t f, flint_rand_t state, flint_bitcnt_t bits, int proved)
 {
-    if (bits <= SMALL_FMPZ_BITCOUNT_MAX)
+    if (bits <= FLINT_BITS)
     {
-        _fmpz_demote(f);
-        *f = n_randprime(state, bits, proved);
+        ulong p = n_randprime(state, bits, proved);
```

Comment:
Why not just ``fmpz_set_ui(f, n_randprime(state, bits, proved));`` ?

Reply:
Performance, but the benefit must be very minimal